### PR TITLE
Table sort headers

### PR DESCRIPTION
**Pull request by joshsadam**
## What does this PR do and why?
_Describe in detail what your merge request does and why._

Added a new view component that uses the ransack object to create sort links in table headers.

## Screenshots or screen recordings
_Screenshots are required for UI changes, and strongly recommended for all other pull requests._

**Project Samples**
image

**Group Samples**
image

## How to set up and validate locally
_Numbered steps to set up and validate the change are strongly suggested._

Currently set up on the project samples page.

## PR acceptance checklist
This checklist encourages us to confirm any changes have been analyzed to reduce risks in quality, performance, reliability, security, and maintainability.

- [x] I have evaluated the [PR acceptance checklist](https://phac-nml.github.io/irida-next/docs/development/development_processes/code_review#acceptance-checklist) for this PR.

### CODE REVIEW

**Comment by deepsidhu85** [comment]
@joshsadam @ericenns Since we are moving the sorting to the table headers, do we need the dropdown for the filtering next to to the search box? I think we should remove it

File: test/system/projects/samples_test.rb [hunk 1]
```diff
@@ -347,21 +347,32 @@ class SamplesTest < ApplicationSystemTestCase
...
+      click_on I18n.t('projects.samples.table.sample')
+
+      assert_selector 'table thead th:nth-child(2) svg.icon-arrow_down'
+      assert_selector 'table#samples-table tbody tr', count: 3
+      within first('tbody') do
+        assert_selector 'tr:first-child td:nth-child(2)', text: @sample3.name
+        assert_selector 'tr:nth-child(2) td:nth-child(2)', text: @sample2.name
+        assert_selector 'tr:last-child td:nth-child(2)', text: @sample1.name
       end
     end
```
Comment (deepsidhu85) [inline test/system/projects/samples_test.rb:377]: Can you also add tests for sorting by the other default columns of the table which have sorting enabled?

**Review by deepsidhu85 — COMMENTED** [review]
This looks good and works as expected. Just a few comments below

**Comment by ericenns** [comment]
> @joshsadam @ericenns Since we are moving the sorting to the table headers, do we need the dropdown for the filtering next to to the search box? I think we should remove it

Yes it needs to removed, I mentioned this to Josh yesterday

File: app/controllers/projects/samples/transfers_controller.rb [hunk 2]
```diff
@@ -3,13 +3,14 @@
...
     # Controller actions for Project Samples Transfer
-    class TransfersController < Projects::ApplicationController
+    class TransfersController < Projects::SamplesController
       respond_to :turbo_stream
       before_action :projects
 
       def new
         authorize! @project, to: :transfer_sample?
 
+        @q = Sample.where(project_id: @project.id).ransack(params[:q])
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply app/controllers/projects/samples/transfers_controller.rb]: Updated in 351f5a0

File: app/controllers/projects/samples/transfers_controller.rb [hunk 3]
```diff
@@ -22,7 +23,8 @@ def create # rubocop:disable Metrics/MethodLength, Metrics/AbcSize
         sample_ids = transfer_params[:sample_ids]
         transferred_samples_ids = ::Samples::TransferService.new(@project, current_user).execute(new_project_id,
                                                                                                  sample_ids)
-        @pagy, @samples = pagy(Sample.where(project_id: @project.id))
+        @q = Sample.where(project_id: @project.id).ransack(params[:q])
```
[the remark this answers is not in the document]
Comment (joshsadam) [reply app/controllers/projects/samples/transfers_controller.rb]: Updated in 351f5a0

File: test/system/projects/samples_test.rb (hunk 1 above)
Comment (joshsadam) [reply test/system/projects/samples_test.rb:413]: Updated in 4841c4c

**Comment by deepsidhu85** [comment]
@joshsadam  found a small spacing issue with the headers. This only seems to be an issue on the Group samples page. Project samples page is fine

**Group samples**
image

**Project samples**
image

**Review by deepsidhu85 — CHANGES_REQUESTED** [review]
Everything is working as expected. Just left a general comment below about a spacing issue with the headers on the Group samples page

**Comment by joshsadam** [comment]
@deepsidhu85 fixed the groups samples table spacing issues n 3653fa8

**Review by deepsidhu85 — DISMISSED** [review]

File: app/controllers/projects/samples/transfers_controller.rb [hunk 4]
```diff
@@ -3,13 +3,14 @@
 module Projects
   module Samples
     # Controller actions for Project Samples Transfer
-    class TransfersController < Projects::ApplicationController
+    class TransfersController < Projects::SamplesController
```
Comment (ericenns) [inline app/controllers/projects/samples/transfers_controller.rb:6]: **nitpick (blocking):** Revert this change and move `load_samples` into `Projects::ApplicationController`.

I think you did this to get access to `load_samples`, but that could be added to the application controller instead. As other wise a lot more logic is being added then is necessary.

**Review by ericenns — CHANGES_REQUESTED** [review]
Looks good other than the inheritance in the `Projects::Samples::TransfersController`

File: app/controllers/projects/samples/transfers_controller.rb (hunk 4 above)
Comment (joshsadam) [reply app/controllers/projects/samples/transfers_controller.rb]: I think this is addressed in e1ef33a

**Review by ericenns — APPROVED** [review]

**Pull request merged**